src/mpcCoreKit.ts

```diff
@@ -872,39 +891,66 @@ export class Web3AuthMPCCoreKit implements ICoreKit, IMPCContext {
     };
   }
 
-  public async sign(
-    data: Buffer,
-    opts?: {
-      hashed?: boolean;
-      secp256k1Precompute?: Secp256k1PrecomputedClient;
-      keyTweak?: BN;
+  public async signECDSA(data: Uint8Array, opts?: { hashed?: boolean; secp256k1Precompute?: Secp256k1PrecomputedClient }) {
+    if (!this.supportedCurveKeyTypes.has(KeyType.secp256k1)) {
+      throw CoreKitError.default(`secp256k1 KeyTYpe is not supported, please configure secp256k1 curve key type `);
     }
-  ): Promise<Buffer> {
-    // this.wasmLib = await this.loadTssWasm();
-    if (this._sigType === "ecdsa-secp256k1") {
-      if (opts?.keyTweak) {
-        throw CoreKitError.default("key tweaking not supported for ecdsa-secp256k1");
-      }
-      const sig = await this.sign_ECDSA_secp256k1(data, opts?.hashed, opts?.secp256k1Precompute);
-      return Buffer.concat([sig.r, sig.s, Buffer.from([sig.v])]);
-    } else if (this._sigType === "ed25519" || this._sigType === "bip340") {
-      if (opts?.hashed) {
-        throw CoreKitError.default(`hashed data not supported for bip340`);
-      } else if (opts?.keyTweak && this._sigType !== "bip340") {
-        throw CoreKitError.default("key tweaking not supported for ed25519");
-      }
+    if (!this.supportedSigTypes.has(SIG_TYPE.ECDSA_SECP256K1)) {
+      throw CoreKitError.default(`ECDSA_SECP256K1 is not supported, please configure tssLib with ECDSA_SECP256K1 signature type `);
+    }
+    const { hashed = false, secp256k1Precompute } = opts || {};
 
-      return this.sign_frost(data, opts?.keyTweak);
+    // TODO: replace buffer to uint8array
+    const sig = await this.sign_ECDSA_secp256k1(Buffer.from(data), hashed, secp256k1Precompute);
+    return Buffer.concat([sig.r, sig.s, Buffer.from([sig.v])]);
+  }
+
+  public async signBip340(data: Uint8Array, opts?: { hashed?: boolean; keyTweak?: BN }) {
```

Comment:
i'm still torn on whether it is better to keep the `sign` function with a `keyType` argument instead? what were the arguments for doing it with two separate functions like here?